### PR TITLE
 ref(ts): Convert contextSummary to Typescript

### PR DESCRIPTION
**Pull request by taylangocmen**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx [hunk 1]
```diff
@@ -13,9 +13,15 @@ import ContextSummaryOS from './contextSummaryOS';
 import ContextSummaryUser from './contextSummaryUser';
 import filterContexts from './filterContexts';
 
+export type Context = {
+  keys: string[];
+  Component: (props: any) => JSX.Element;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx:18]: I don't like much the any here, but I think that it's fine for now.. we should refactor this component in the future :)

File: src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx [hunk 2]
```diff
@@ -32,33 +38,33 @@ const KNOWN_CONTEXTS = [
...
-    event: SentryTypes.Event.isRequired,
-  };
+type Props = {
+  event: Event;
+};
 
+class ContextSummary extends React.Component<Props> {
   render() {
-    const evt = this.props.event;
+    const {event} = this.props;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx:47]: 👏

File: src/sentry/static/sentry/app/components/events/contextSummary/filterContexts.tsx [hunk 3]
```diff
@@ -1,10 +1,14 @@
-function filterContexts(event, context) {
+import {Event} from 'app/types';
+
+import {Context} from './contextSummary';
+
+function filterContexts(event: Event, context: Context) {
   // if the operating system is macOS, we want to hide devices called "Mac" which don't have any additional info
   if (context.keys.includes('device')) {
-    const {model, arch, data_id} = event.contexts?.device || {};
+    const {model, arch} = event.contexts?.device || {};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/contextSummary/filterContexts.tsx:8]: are you sure about removing "data_id"? wasn't it being used?

File: src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx:18]: Could we add a `TODO(ts)` here?

File: src/sentry/static/sentry/app/components/events/contextSummary/filterContexts.tsx (hunk 3 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/components/events/contextSummary/filterContexts.tsx:8]: I looked in `snuba/events` and test pysnaps but couldn't find this field anywhere

File: src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx (hunk 1 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/components/events/contextSummary/contextSummary.tsx:18]: 👍

**Review by markstory — APPROVED** [review]

**Pull request merged**